hbase-server/src/main/java/org/apache/hadoop/hbase/master/MasterRpcServices.java

```diff
@@ -1706,12 +1707,23 @@ public SnapshotResponse snapshot(RpcController controller,
       // get the snapshot information
       SnapshotDescription snapshot = SnapshotDescriptionUtils.validate(
         request.getSnapshot(), server.getConfiguration());
-      server.snapshotManager.takeSnapshot(snapshot);
-
       // send back the max amount of time the client should wait for the snapshot to complete
       long waitTime = SnapshotDescriptionUtils.getMaxMasterTimeout(server.getConfiguration(),
         snapshot.getType(), SnapshotDescriptionUtils.DEFAULT_MAX_WAIT_TIME);
-      return SnapshotResponse.newBuilder().setExpectedTimeout(waitTime).build();
+
+      SnapshotResponse.Builder builder = SnapshotResponse.newBuilder().setExpectedTimeout(waitTime);
+
+      // just to pass the unit tests for all 3.x versions,
+      // the minimum version maybe needs to be modified later
+      if (VersionInfoUtil.currentClientHasMinimumVersion(2, 10)) {
```

Comment:
I think here we could always use the same logic? Just always return the procId, it is the client's choice to whether to make use of it. And we could just test whether we have nonceGroup and nonce directly, so we do not need to test the client version.

Reply:
Ok. That's a good idea. I will address it ASAP.